modules/Users/User.php

```diff
@@ -653,11 +653,14 @@ public function save($check_notify = false)
         parent::save($check_notify);
 
         // User Profile specific save for Email addresses
-        if (!$this->emailAddress->saveAtUserProfile($_REQUEST)) {
-            $GLOBALS['log']->error('Email address save error');
-            return false;
+        if ($this->emailAddress->isUserProfileEditViewPageSaveAction($_REQUEST)){
+            if (!$this->emailAddress->saveAtUserProfile($_REQUEST)) {
+                $GLOBALS['log']->error('Email address save error');
+                return false;
+            }
```

Comment:
You can merge these two ```if``` statements together. While there I would also suggest replacing the ```$GLOBALS['log']``` with the ```LoggerManager::getLogger()->error``` leaving you with:
```php
    if ($this->emailAddress->isUserProfileEditViewPageSaveAction($_REQUEST) && 
        !$this->emailAddress->saveAtUserProfile($_REQUEST)) {
        LoggerManager::getLogger()->error('Email address save error');
        return false;
    }
```